File: app/src/main/java/br/com/yourapp/pokeapi/models/PokemonInfo.java
package br.com.yourapp.pokeapi.models;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.List;

public class PokemonInfo implements Serializable {
    @SerializedName("abilities")
    public List<Abilities> abilities;

    @SerializedName("base_experience")
    public String base_experience;

    @SerializedName("forms")
    public List<Forms> forms;

    @SerializedName("game_indices")
    public List<GameIndex> game_indices;

    @SerializedName("height")
    public int height;

    @SerializedName("weight")
    public int weight;

    @SerializedName("id")
    public int id;

    @SerializedName("id_default")
    public boolean id_default;

    @SerializedName("location_area_encounters")
    public String location_area_encounters;

    @SerializedName("name")
    public String name;

    @SerializedName("order")
    public int order;

    @SerializedName("sprites")
    public Sprites sprites;

    public class Sprites {
        @SerializedName("back_default")
        public String back_default;

        @SerializedName("back_female")
        public String back_female;

        @SerializedName("back_shiny")
        public String back_shiny;

        @SerializedName("back_shiny_female")
        public String back_shiny_female;

        @SerializedName("front_default")
        public String front_default;

        @SerializedName("front_female")
        public String front_female;

        @SerializedName("front_shiny")
        public String front_shiny;

        @SerializedName("front_shiny_female")
        public String front_shiny_female;

        @SerializedName("other")
        public Other other;
    }

    public class Other {
        @SerializedName("dream_world")
        public DreamWorld dream_world;

        @SerializedName("official-artwork")
        public OfficialArtwork official_artwork;
    }

    public class DreamWorld {
        @SerializedName("front_default")
        public String front_default;

        @SerializedName("front_female")
        public String front_female;
    }

    public class OfficialArtwork {
        @SerializedName("front_default")
        public String front_default;
    }
}

File: app/src/main/java/br/com/yourapp/pokeapi/models/PokemonInfo.kt
package br.com.yourapp.pokeapi.models

import com.google.gson.annotations.SerializedName
import java.io.Serializable

class PokemonInfo : Serializable {
    @SerializedName("abilities")
    var abilities: List<Abilities>? = null

    @SerializedName("base_experience")
    var base_experience: String? = null

    @SerializedName("forms")
    var forms: List<Forms>? = null

    @SerializedName("game_indices")
    var game_indices: List<GameIndex>? = null

    @SerializedName("height")
    var height = 0

    @SerializedName("weight")
    var weight = 0

    @SerializedName("id")
    var id = 0

    @SerializedName("id_default")
    var id_default = false

    @SerializedName("location_area_encounters")
    var location_area_encounters: String? = null

    @SerializedName("name")
    var name: String? = null

    @SerializedName("order")
    var order = 0

    @SerializedName("sprites")
    var sprites: Sprites? = null

    inner class Sprites : Serializable {
        @SerializedName("back_default")
        var back_default: String? = null

        @SerializedName("back_female")
        var back_female: String? = null

        @SerializedName("back_shiny")
        var back_shiny: String? = null

        @SerializedName("back_shiny_female")
        var back_shiny_female: String? = null

        @SerializedName("front_default")
        var front_default: String? = null

        @SerializedName("front_female")
        var front_female: String? = null

        @SerializedName("front_shiny")
        var front_shiny: String? = null

        @SerializedName("front_shiny_female")
        var front_shiny_female: String? = null

        @SerializedName("other")
        var other: Other? = null
    }

    @SerializedName("types")
    var types: List<Types>? = null

    inner class Types : Serializable {
        @SerializedName("slot")
        var slot: String? = null

        @SerializedName("type")
        var type: Type? = null
    }

    inner class Type : Serializable {
        @SerializedName("name")
        var name: String? = null

        @SerializedName("url")
        var url: String? = null
    }

    inner class Other : Serializable {
        @SerializedName("dream_world")
        var dream_world: DreamWorld? = null

        @SerializedName("official-artwork")
        var official_artwork: OfficialArtwork? = null
    }

    inner class DreamWorld : Serializable {
        @SerializedName("front_default")
        var front_default: String? = null

        @SerializedName("front_female")
        var front_female: String? = null
    }

    inner class OfficialArtwork : Serializable {
        @SerializedName("front_default")
        var front_default: String? = null
    }
}